pkg/inventory/inventory-client.go

```diff
@@ -350,9 +361,15 @@ func (cic *ClusterInventoryClient) applyInventoryObj(obj *unstructured.Unstructu
 	}
 
 	klog.V(4).Infof("replacing inventory object: %s/%s", obj.GetNamespace(), obj.GetName())
-	_, err = cic.dc.Resource(mapping.Resource).Namespace(obj.GetNamespace()).
-		Update(context.TODO(), obj, metav1.UpdateOptions{})
-	return err
+	var updatedObj *unstructured.Unstructured
+	if updatedObj, err = cic.dc.Resource(mapping.Resource).Namespace(obj.GetNamespace()).
+		Update(context.TODO(), obj, metav1.UpdateOptions{}); err != nil {
+		return err
+	}
+	if updatedObj != nil {
+		cic.invResourceVersion = updatedObj.GetResourceVersion()
```

Comment:
Why not just return the updatedObj and copy the ResourceVersion in the caller?

Reply:
done